### PR TITLE
upgrading gopowerstore dependency to v1.14.0

### PR DESCRIPTION
**Pull request by prajwalpatil25**
# Description
Upgraded gopowerstore dependency version to v1.14.0

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# Checklist:

- [X] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [X] I have verified that new and existing unit tests pass locally with my changes
- [X] I have not allowed coverage numbers to degenerate
- [X] I have maintained at least 90% code coverage
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [ ] Acceptance tests

### CODE REVIEW

**Comment by prajwalpatil25** [comment]
run e2e test

**Comment by terraform-providers-svc** [comment]
Can one of the admins verify this patch?

**Review by Sakshi-dell — APPROVED** [review]

**Review by doriac11 — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -73,7 +73,7 @@ require (
 	github.com/posener/complete v1.2.3 // indirect
 	github.com/russross/blackfriday v1.6.0 // indirect
 	github.com/shopspring/decimal v1.3.1 // indirect
-	github.com/sirupsen/logrus v1.8.1 // indirect
+	github.com/sirupsen/logrus v1.9.3 // indirect
 	github.com/spf13/cast v1.5.0 // indirect
 	github.com/stretchr/testify v1.8.1
```
Comment (Krunal-Thakkar) [inline go.mod:78]: Can we update testify to latest version?

Comment (prajwalpatil25) [reply go.mod:78]: I think can be part of another PR.

**Pull request merged**